### ISSUE TITLE
Dont work 2.0 version

### ISSUE THREAD

**Issue by Netrake** [dotStart/Beacon#1]
Version 1.1 Work

**Comment by dotStart** [comment]
That's not particularly helpful. What does not work?

* Does the application start?
* Is an error reported by the application?
* Have you tried launching the jar version from the console?

1.x differs entirely from 2.x and thus does not compare.

**Comment by Netrake** [comment]
Application not start
no
yes

**Comment by dotStart** [comment]
Without knowing the internal error there's no sensible way to debug the issue.

Please paste the full message here if the application provides an error dialogue. If not:

1. [Download a jar copy of the application](https://github.com/dotStart/Beacon/releases/latest/download/Beacon.jar) if you haven't done so yet
2. Run the application via `java -jar Beacon.jar` from a command line (preferably git bash if available)
3. Copy the *entire* output and paste it in a text file
4. Attach the text file here

**Comment by ZenithLord** [comment]
Hi Johannes,

I have the same problem; I'd like to provide some information that the
previous person did not provide.
 - The application starts completely fine (Beacon 2.0 that is, written on
Kotlin). However, the loading screen shows and never goes away. The main
GUI to show games and ports does not show.
 - I downloaded previous versions of Beacon to test what happened; turns
out, the most recent version of Beacon after Beacon 2 functioned perfectly
fine!
 - Right now, I forgot the other issues, but I'll try to notify you as soon
as possible. Please let me know if there's anything I've omitted and I can
do.

Yours,
Matt

**Comment by dotStart** [comment]
Thanks for the additional info.

Since it is incredibly hard to debug this issue as-is, I just released v2.1.0 which flushes the application logs to disk without requiring users to launch the application from the command line. As such, I'd recommend to [update to 2.1](https://github.com/dotStart/Beacon/releases/tag/v2.1.0) and attach the log files in question here.

As of the latest release, the log file may be found at:
* `%APPDATA%\Beacon\log\latest.log` on **Windows**
*`~/.beacon/log/latest.log` for **Linux, Mac OS and other NIX based systems**

**Comment by KirillPUH** [comment]
I have this problem too.
[log.txt](https://github.com/dotStart/Beacon/files/3443566/log.txt)

**Comment by dotStart** [comment]
> I have this problem too.
> Aplication starts, but it doesn't go further than the boot screen.
> [log.txt](https://github.com/dotStart/Beacon/files/3443566/log.txt)

According to the log you are running Java 8 (or older) which is not compatible with Beacon anymore. You need Java 9 or higher installed in your path.

That being said: Clearly, the Windows wrapper is not doing its job at the moment (there should be a warning regarding this). The same goes for the jar itself, of course.

**Comment by dotStart** [comment]
Little update on this issue:

Another user has agreed to send me a copy of their log privately. As it turns out, there is most likely an exception being thrown on the JavaFX thread which is not being reported accordingly by the application (instead the event loop remains alive while the application is no longer capable of proceeding). This issue has been addressed with v2.1.2 (e.g. issues like this should now be logged correctly and reported via an error dialogue if possible).

That being said: The actual underlying issue is most likely still present. It is simply being reported correctly now (hopefully). As I cannot reproduce this issue yet, I cannot validate this though ...

**Comment by dotStart** [comment]
This issue seems to be caused by [one of our dependencies](https://github.com/jfoenixadmin/JFoenix/issues/976) on Java 12 (and newer) and should be resolved with v2.1.4 (the feature in question has been replaced).

As such, I'll consider this issue to be fixed for now. If anything related to this issue comes up, feel free to re-open the issue.

For the sake of transparency, here's the crash report that tipped me off on this particular issue:

```
Beacon Error Report
===================
Version: 2.1.2
Date: 2019-07-30T18:50:35.1780489-04:00
Java Version: 12.0.2
Java Architecture: amd64
Operating System: Windows 10 (10.0)

Stack Trace:
------------
java.lang.NullPointerException
at com.jfoenix.skins.JFXTextFieldSkin.updateTextPos(JFXTextFieldSkin.java:109)
at com.jfoenix.skins.JFXTextFieldSkin.layoutChildren(JFXTextFieldSkin.java:93)
at javafx.scene.control.Control.layoutChildren(Control.java:601)
at javafx.scene.Parent.layout(Parent.java:1204)
at javafx.scene.Parent.layout(Parent.java:1211)
at javafx.scene.Parent.layout(Parent.java:1211)
at javafx.scene.Parent.layout(Parent.java:1211)
at javafx.scene.Parent.layout(Parent.java:1211)
at javafx.scene.Parent.layout(Parent.java:1211)
at javafx.scene.Scene.doLayoutPass(Scene.java:576)
at javafx.scene.Scene.preferredSize(Scene.java:1748)
at javafx.scene.Scene$2.preferredSize(Scene.java:393)
at com.sun.javafx.scene.SceneHelper.preferredSize(SceneHelper.java:66)
at javafx.stage.Window$12.invalidated(Window.java:1086)
at javafx.beans.property.BooleanPropertyBase.markInvalid(BooleanPropertyBase.java:110)
at javafx.beans.property.BooleanPropertyBase.set(BooleanPropertyBase.java:145)
at javafx.stage.Window.setShowing(Window.java:1174)
at javafx.stage.Window.show(Window.java:1189)
at javafx.stage.Stage.show(Stage.java:273)
at tv.dotstart.beacon.controller.SplashController$initialize$2.invoke(SplashController.kt:78)
at tv.dotstart.beacon.controller.SplashController$initialize$2.invoke(SplashController.kt:40)
at tv.dotstart.beacon.preload.Preloader$invoke$1$5.run(Preloader.kt:120)
at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:389)
at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
at java.base/java.lang.Thread.run(Thread.java:835) 
```
